apps/entity-service/src/entities/processors/financial-report.processor.ts

```diff
@@ -23,6 +29,57 @@ export class FinancialReportProcessor extends ReportProcessor<
   readonly LIGHT_DTO = FinancialReportLightDto;
   readonly FULL_DTO = FinancialReportFullDto;
 
+  /**
+   * Specific method for FinancialReport custom logic. This is called automatically when nothingToReport is updated
+   */
+  protected async processFinancialReportSpecificLogic(model: FinancialReport): Promise<void> {
+    const organisation = await Organisation.findByPk(model.organisationId);
+    if (organisation == null) {
+      console.warn(`Organisation not found for FinancialReport ${model.uuid}`);
+      return;
+    }
+
+    if (model.finStartMonth != null || model.currency != null) {
+      const updateData: Partial<Organisation> = {};
+      if (model.finStartMonth != null) updateData.finStartMonth = model.finStartMonth;
+      if (model.currency != null) updateData.currency = model.currency;
+
+      await organisation.update(updateData);
+    }
+
+    const reportIndicators = await FinancialIndicator.financialReport(model.id).findAll();
+    const existingOrgIndicators = await FinancialIndicator.organisation(organisation.id).findAll();
+
+    const orgIndicatorMap = new Map<string, FinancialIndicator>();
+    existingOrgIndicators.forEach(indicator => {
+      const key = `${indicator.year}-${indicator.collection}`;
+      orgIndicatorMap.set(key, indicator);
+    });
+
+    for (const reportIndicator of reportIndicators) {
```

Comment:
Is this likely to be a lot of indicators? Any time I see `create` in a loop, I want to move it to a [bulk create](https://sequelize.org/docs/v6/core-concepts/model-querying-basics/#creating-in-bulk). Since this look covers both create and update then you'd need to split into bulk create and [bulk update](https://sequelize.org/docs/v7/querying/update/#updating-in-bulk). I would only bother if this is likely to be more than 10 or in any given request.

Reply:
Oh shoot, I didn't realize how the bulk update works. I was thinking it was more like an upsert where you can update multiple different rows with different values at once, but that doesn't seem to be the case. I'll go ahead and approve, but in the future, I would have left your code similar to how it was before for the update path. The only change would be to push the update promises to an array instead of `await`ing them inline, and then just await the Promise.all after the loop is complete.